test/integration/node/lifecycle_test.go

```diff
@@ -153,11 +193,12 @@ func TestEvictionForNoExecuteTaintAddedByUser(t *testing.T) {
 			if err := testutils.AddTaintToNode(cs, nodes[nodeIndex].Name, v1.Taint{Key: "CustomTaintByUser", Effect: v1.TaintEffectNoExecute}); err != nil {
 				t.Errorf("Failed to taint node in test %s <%s>, err: %v", name, nodes[nodeIndex].Name, err)
 			}
-
 			err = wait.PollImmediate(time.Second, time.Second*20, testutils.PodIsGettingEvicted(cs, testPod.Namespace, testPod.Name))
-			if err != nil {
+			if ((test.enableSeparateTaintManager && !test.startStandaloneTaintManager && wait.Interrupted(err)) ||
+				(!test.enableSeparateTaintManager && err != nil)) != !test.wantPodEvicted {
```

Comment:
This may read better:

```go
if test.wantPodEvicted {
    // expect the pod to be evicted
} else {
    // expect the pod to be not evicted
}
```

Reply:
Done this in 080643d665e